### PR TITLE
Fixes #11836: sharedfile_to_node reports repaired even if it fails 

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/11836

### CODE REVIEW

**Comment by peckpeck** [comment]
Indeed, all our curl commands should be based on a single option list

**Comment by matya** [comment]
But that's a different story

**Comment by peckpeck** [comment]
Commit modified

File: tree/30_generic_methods/sharedfile_to_node.cf [hunk 1]
```diff
@@ -118,6 +118,11 @@ bundle agent sharedfile_to_node(target_uuid, file_id, file_path, ttl)
 
       "url_ttl"           string => execresult("echo ${ttl} | sed 's/ /+/g'", "useshell");
 
+    inventory_proxy_defined::
+      "curl_proxy"        string => "";
+    !inventory_proxy_defined::
+      "curl_proxy"        string => "${node.properties[inventory_proxy]}";
```
Comment (amousset) [inline tree/30_generic_methods/sharedfile_to_node.cf]: * Why inventory_proxy?
* This really needs to be documented, in a section about "special variables and parameters"

Comment (peckpeck) [reply tree/30_generic_methods/sharedfile_to_node.cf]: We will do https://www.rudder-project.org/redmine/issues/12100 instead

**Comment by peckpeck** [comment]
Commit modified

File: tree/30_generic_methods/sharedfile_to_node.cf [hunk 2]
```diff
@@ -156,14 +156,13 @@ bundle agent sharedfile_to_node(target_uuid, file_id, file_path, ttl)
         ifvarclass => "!${canonified_path}_signed";
 
     _stdlib_path_exists_curl.pass2::
-      "${paths.path[curl]}"
-        args => "${url}?hash=${hash} -k -1 -L --head -o /dev/null -s -f ", 
+      "code=`${paths.path[curl]} ${url}?hash=${hash} --insecure --tlsv1 --location --head --output /dev/null --silent --fail --write-out %{http_code}` || [ $code -eq 404 ] && exit 22 || exit 254"
         contain => in_shell,
-        classes => classes_generic_return_single_code_two("${old_class_prefix}_exist", "${class_prefix}_exist", "0", "22", "999"), # 22 = http 400 or above
+        classes => classes_generic_return_single_code_two("${old_class_prefix}_exist", "${class_prefix}_exist", "0", "22", "254"), # 22 = http 400 or above
```
Comment (amousset) [inline tree/30_generic_methods/sharedfile_to_node.cf]: comment needs to be updated

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**